### PR TITLE
Make first cuda op build with cuda context

### PR DESCRIPTION
**Pull request by RandySheriffH**
Make first cuda op build with a working cuda context.

### CODE REVIEW

File: includes/custom_op_lite.h [hunk 1]
```diff
@@ -418,6 +418,21 @@ struct Variadic : public TensorBase {
   TensorPtrs tensors_;
 };
 
+#ifdef USE_CUDA
+
+struct CudaContext {
+  void Init(const OrtW::CustomOpApi& api, const OrtKernelContext& ctx) {
+    const auto& ort_api = api.GetOrtApi();
+    ort_api.KernelContext_GetResource(&ctx, 1, 10000, &cuda_stream);
```
Comment (wenbingl) [inline includes/custom_op_lite.h:426]: what's 10000?

File: includes/custom_op_lite.h [hunk 2]
```diff
@@ -418,6 +418,21 @@ struct Variadic : public TensorBase {
...
+    if (!cuda_stream) {
+      ORTX_CXX_API_THROW("Failed to fetch cuda stream from context", ORT_RUNTIME_EXCEPTION);
```
Comment (wenbingl) [inline includes/custom_op_lite.h:428]: Given now API is fallible, returning a status is a better now.

File: CMakeLists.txt [hunk 3]
```diff
@@ -816,7 +816,6 @@ endif()
 if(OCOS_USE_CUDA)
   target_include_directories(ortcustomops PUBLIC ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
   target_link_libraries(extensions_shared PUBLIC cudart cublas cufft)
-  set_target_properties(ocos_operators PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
```
Comment (wenbingl) [inline CMakeLists.txt:819]: Do you have some insight about which way is better?

Comment (RandySheriffH) [reply CMakeLists.txt:819]: The CUDA_SEPARABLE_COMPILATION cause link issues.
Removing it  helper the build pass.

File: includes/custom_op_lite.h (hunk 2 above)
Comment (RandySheriffH) [reply includes/custom_op_lite.h:428]: The ::Init() is to be called during “CreateTuple(...)”, which return a **tuple** to a compute function.
We cannot return a status upwards to notify the failture.

**Review by wenbingl — DISMISSED** [review]

File: includes/custom_op_lite.h (hunk 1 above)
Comment (RandySheriffH) [reply includes/custom_op_lite.h]: Added an enum to specify the resource type.

**Review by wenbingl — APPROVED** [review]

**Pull request merged**